src/aws-client/Client.js

```diff
@@ -0,0 +1,366 @@
+'use strict'
+
+const path = require('path')
+const fs = require('fs/promises')
+const { Piscina } = require('piscina')
+const { Agent, request } = require('undici')
+const { xml2js } = require('xml-js')
+const { BufferList } = require('bl')
+const { unmarshall } = require('@aws-sdk/util-dynamodb')
+const { serializeError } = require('../logging')
+const sleep = require('util').promisify(setTimeout)
+
+const signerWorker = new Piscina({
+  filename: path.resolve(__dirname, './signer-worker.js'),
+  idleTimeout: Math.pow(2, 31) - 1
+})
+
+/**
+ * s3 requests are multi region
+ * dynamo requests point to a single region, defined in `dynamoOptions.region`
+ * @see https://docs.aws.amazon.com/index.html
+ */
+class Client {
```

Comment:
Any particular reason for using a `class`? I hope I'm not unlocking spaces vs tabs debate, but avoid `class` es as much as possible 😉